### ISSUE TITLE
Find shape by xpath

### ISSUE THREAD

**Issue by dorvak** [scanny/python-pptx#224]
Hi,

first of all: Great library! here is my question:

Is there any option to find a shape by its name? I' could locate it using the spTree & xpath, but this returns just the XML without any access to all the nice wrappers and the top-level API. Such an option would be super helpful.

Thx in advance

**Comment by YouGovGit** [comment]
Some analysis on shape id and name here: https://python-pptx.readthedocs.io/en/latest/dev/analysis/features/shp-shapes.html

**Comment by scanny** [comment]
@dorvak - Were you thinking within a slide? Or across the entire presentation?

Either way, this would be easily accomplished with something like this:

```python
def find_shape_by_name(shapes, name):
    for shape in shapes:
        if shape.name == name:
            return shape
    return None
```

Note that shape names are not a unique identifier, so this would only return the first one.

Let us know more specifics about your use cases and we can make this a feature request for it. Maybe something like Shapes.get_by_name() or Shapes.get_all_by_name().

**Comment by dorvak** [comment]
Yes, something like Shapes.get_by_name() or Shapes.get_all_by_name() would be great. What I'm currently trying is to fill an existing pptx (the blueprint) with data from other sources (f.e. a database), using meaningfull shape names as placeholders. 
Ideally, one could write a mapping table (datasource->shapename/xpath-selector for a shape; f.e. as a definition file in xml or json) to fill out such blueprint-pptx's. 

I tried it using lxml's xpath functions (search within a slide):
`presentation.slides[1].spTree.xpath(".//*[@name='Kurzkonzept']")`

This works, but it will only retrieve a list of (lxml) Element-Objects that does not provide the functionality of the python-pptx API. Maybe there is a really easy way to "convert" it back to an python-pptx object, but I could not figure out how to do that.

**Comment by scanny** [comment]
What does an XPath search give you that a helper function like the one I suggested wouldn't?

**Comment by dorvak** [comment]
Flexibility, I guess. You could search for values, classes etc. using XPath.

**Comment by scanny** [comment]
Alright, well if you really wanted to do it that way, this should do what you're looking for:

```python
from pptx.shapes.shapetree import SlideShapeFactory

def iter_matching_shapes(slide, xpath_expr):
    shapes, spTree = slide.shapes, slide._spTree
    matching_elements = spTree.xpath(xpath_expr)
    for element in matching_elements:
        yield SlideShapeFactory(element, shapes)
```
